DirectProgramming/DPC++FPGA/ReferenceDesigns/db/src/query9/query9_kernel.cpp

```diff
@@ -0,0 +1,656 @@
+#include <array>
+#include <stdio.h>
+#include <type_traits>
+#include <vector>
+
+#include "query9_kernel.hpp"
+#include "pipe_types.hpp"
+
+#include "../db_utils/Accumulator.hpp"
+#include "../db_utils/LikeRegex.hpp"
+#include "../db_utils/MapJoin.hpp"
+#include "../db_utils/MergeJoin.hpp"
+#include "../db_utils/Misc.hpp"
+#include "../db_utils/ShannonIterator.hpp"
+#include "../db_utils/Tuple.hpp"
+#include "../db_utils/Unroller.hpp"
+#include "../db_utils/fifo_sort.hpp"
+
+using namespace std::chrono;
```

Comment:
Can you use dpc_common.hpp timers?

Reply:
Not flexible enough, don't expose the ability to cast to different units (i.e. milliseconds). Results in ugly extra conversion.